reccmp/tools/project.py

```diff
@@ -0,0 +1,448 @@
+                #endif
+            """
+            )
+    return textwrap.dedent(
+        f"""\
+        #include "{hpp_path.name}"
+
+        // FUNCTION: {target_id} 0x10000000
+        SomeClass::~SomeClass() {{
+        }}
+
+        // GLOBAL: {target_id} 0x10102000
+        // STRING: {target_id} 0x10101f00
+        const char* g_globalString = "A global string";
+
+        {entry_function}
+    """
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Project management", allow_abbrev=False
+    )
+    parser.set_defaults(action=None)
+    parser.add_argument(
+        "--version", action="version", version=f"%(prog)s {reccmp.VERSION}"
+    )
+    parser.add_argument(
+        "-C",
+        type=Path,
+        dest="cwd",
+        metavar="<path>",
+        default=Path.cwd(),
+        help="Run as if %(prog)s was started in %(metavar)s",
+    )
+    subparsers = parser.add_subparsers()
+
+    create_parser = subparsers.add_parser("create")
+    create_parser.set_defaults(action="CREATE")
+    create_parser.add_argument(
+        "--originals",
+        type=Path,
+        nargs="+",
+        metavar="ORIGINAL",
+        dest="create_originals",
+        required=True,
+        help="Path(s) of original executable(s)",
+    )
+    create_parser.add_argument(
+        "--path",
+        metavar="<project-directory>",
+        dest="create_directory",
+        type=Path,
+        default=Path.cwd(),
+        help="Location where to create reccmp project",
+    )
+    create_parser.add_argument(
+        "--cmake-project",
+        action="store_true",
+        dest="create_cmake",
+        help="Create minimal CMake project",
+    )
+
+    detect_parser = subparsers.add_parser("detect")
+    detect_parser.set_defaults(action="DETECT")
+    detect_parser.add_argument(
+        "--search-path",
+        nargs="+",
+        dest="detect_search_path",
+        type=Path,
+        metavar="<path>",
+        default=[Path.cwd()],
+        help="Directory in which to look for original binaries",
+    )
+    detect_parser.add_argument(
+        "--what",
+        choices=(DetectWhat.ORIGINAL, DetectWhat.RECOMPILED),
+        type=DetectWhat,
+        default=DetectWhat.ORIGINAL,
+        dest="detect_what",
+        help="Detect original or recompiled binaries (default is original)",
+    )
+
+    argparse_add_logging_args(parser=parser)
+
+    args = parser.parse_args()
+
+    argparse_parse_logging(args=args)
+
+    if args.action == "CREATE":  # FIXME: use enum or callback function
+        return create_project(
+            project_directory=args.create_directory,
+            original_paths=args.create_originals,
+            cmake=args.create_cmake,
+        )
+
+    if args.action == "DETECT":  # FIXME: use enum or callback function
+        project = RecCmpProject.from_directory(Path.cwd())
+        if not project:
+            parser.error("Cannot find reccmp project. Run %(prog)s create first.")
```

Comment:
This doesn't support % replacement and doesn't appear to recognize `%(prog)s` so I think this should be:
```python
parser.error(f"Cannot find reccmp project. Run {parser.prog} create first.")
```